File: FrontEnd-Mobile/app/src/main/java/com/it342/projectmanagementsystem/models/TimestampObject.java
package com.it342.projectmanagementsystem.models;

import com.google.gson.annotations.SerializedName;

public class TimestampObject {
    @SerializedName("seconds")
    private long seconds;
    
    @SerializedName("nanos")
    private int nanos;

    public static TimestampObject fromMillis(long millis) {
        TimestampObject obj = new TimestampObject();
        obj.seconds = millis / 1000;
        obj.nanos = (int) ((millis % 1000) * 1_000_000);
        return obj;
    }

    public long toMillis() {
        return seconds * 1000 + nanos / 1_000_000;
    }

    public long getSeconds() {
        return seconds;
    }

    public void setSeconds(long seconds) {
        this.seconds = seconds;
    }

    public int getNanos() {
        return nanos;
    }

    public void setNanos(int nanos) {
        this.nanos = nanos;
    }
} 

File: FrontEnd-Mobile/app/src/main/java/com/it342/projectmanagementsystem/models/TimestampObject.kt
package com.it342.projectmanagementsystem.models

import java.util.concurrent.TimeUnit

data class TimestampObject(
    val seconds: Long,
    val nanos: Int
) {
    companion object {
        fun fromMillis(millis: Long): TimestampObject {
            val seconds = millis / 1000
            val nanos = ((millis % 1000) * 1_000_000).toInt()
            return TimestampObject(seconds, nanos)
        }
    }

    fun toMillis(): Long {
        return TimeUnit.SECONDS.toMillis(seconds) +
                TimeUnit.NANOSECONDS.toMillis(nanos.toLong())
    }
} 